pkg/config/sink.go

```diff
@@ -62,6 +64,16 @@ func (s *SinkConfig) validate(enableOldValue bool) error {
 			}
 		}
 	}
+	for _, rule := range s.DispatchRules {
+		if rule.DispatcherRule != "" && rule.PartitionRule != "" {
+			log.Error("dispatcher and partition cannot be configured both", zap.Any("rule", rule))
+			return cerror.WrapError(cerror.ErrSinkInvalidConfig,
+				errors.New(fmt.Sprintf("dispatcher and partition cannot be configured both for rule:%v", rule)))
+		}
+		if rule.DispatcherRule != "" {
```

Comment:
Also test for this logic. 
Maybe it would be better to add a comment explaining why we did this?

Reply:
Ditto, I have covered this scenario in `TestReplicaConfigValidate`.

Reply:
For the comment, i will add it later.